soc/riscv/common/riscv-privileged/soc_irq.S

```diff
@@ -59,3 +71,59 @@ SECTION_FUNC(exception.other, __soc_is_irq)
 not_interrupt:
 	/* return */
 	ret
+
+#if defined(CONFIG_NUCLEI_ECLIC) && !defined(CONFIG_RISCV_VECTORED_MODE)
```

Comment:
can this be moved to some `ECLIC` specific place?

Reply:
done. I added a new file intc_nuclei_eclic.S (in the same folder as intc_nuclei_eclic.c) and moved the hooks there.